### ISSUE TITLE
Kernel module issue

### ISSUE THREAD

**Issue by korithi** [sipwise/rtpengine#1564]
Hello,

I just installed the rtpengine 10.5 via this link: https://dfx.at/rtpengine/
System: debian 11 kernel 5.10.0-10-amd64
Status of rtpengine:
● rtpengine-daemon.service - NGCP RTP/media Proxy Daemon
     Loaded: loaded (/lib/systemd/system/rtpengine-daemon.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2022-11-08 15:55:51 EST; 18min ago
    Process: 535 ExecStartPre=/usr/sbin/rtpengine-iptables-setup start (code=exited, status=0/SUCCESS)
   Main PID: 595 (rtpengine)
      Tasks: 21 (limit: 9506)
     Memory: 30.6M
        CPU: 934ms
     CGroup: /system.slice/rtpengine-daemon.service
             └─595 /usr/bin/rtpengine -f -E --no-log-timestamps --pidfile /run/rtpengine/rtpengine-daemon.pid --config-file /etc/rtpengine/rtpengine.conf

Nov 08 15:55:50 webrtc rtpengine-iptables-setup[568]: modprobe: FATAL: Module xt_RTPENGINE not found in directory /lib/modules/5.10.0-10-amd64
Nov 08 15:55:50 webrtc rtpengine-iptables-setup[583]: iptables v1.8.7 (nf_tables): unknown option "--id"
Nov 08 15:55:50 webrtc rtpengine-iptables-setup[583]: Try `iptables -h' or 'iptables --help' for more information.
Nov 08 15:55:50 webrtc rtpengine-iptables-setup[593]: ip6tables v1.8.7 (nf_tables): unknown option "--id"
Nov 08 15:55:50 webrtc rtpengine-iptables-setup[593]: Try `ip6tables -h' or 'ip6tables --help' for more information.
Nov 08 15:55:51 webrtc rtpengine[595]: INFO: [crypto] Generating new DTLS certificate
Nov 08 15:55:51 webrtc rtpengine[595]: ERR: [core] FAILED TO CREATE KERNEL TABLE 0 (No such file or directory), KERNEL FORWARDING DISABLED
Nov 08 15:55:51 webrtc rtpengine[595]: INFO: [core] Startup complete, version 10.5.2.7-1~bpo11+1
Nov 08 15:55:51 webrtc rtpengine[595]: INFO: [http] Websocket listener thread running
Nov 08 15:55:51 webrtc systemd[1]: Started NGCP RTP/media Proxy Daemon.

Can you please help and let me know why it cannot find the module? If this is not the right way how should I install the right version of rtpenginge for this kernel that I have?
Thanks

**Comment by rfuchs** [comment]
Did you install the `-iptables` package?

**Comment by korithi** [comment]
Yes, iptables were installed before installing rtpengine. Version (v1.8.7)

**Comment by rfuchs** [comment]
No, I'm talking about the `rtpengine-iptables` package. Also `rtpengine-kernel-dkms` seems to be missing.

**Comment by korithi** [comment]
Both of them as per below:

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
rtpengine-iptables is already the newest version (10.5.2.7-1~bpo11+1).
rtpengine-iptables set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
---------------------------------------------------------
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
rtpengine-kernel-dkms is already the newest version (10.5.2.7-1~bpo11+1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

**Comment by rfuchs** [comment]
What's the output of `dpkg-reconfigure rtpengine-kernel-dkms` (possibly with `sudo`) ?

**Comment by korithi** [comment]
DKMS: build completed.

xt_RTPENGINE.ko:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/5.10.0-19-amd64/updates/dkms/

depmod...

DKMS: install completed.

Now that I am seeing it there is another library there and as it seems it is being installed at the newest one but when the service starts it search it on the older one.
Libraries as below
ls -l /lib/modules/
total 8
drwxr-xr-x 3 root root 4096 Oct 25 10:16 5.10.0-10-amd64
drwxr-xr-x 3 root root 4096 Nov  8 16:53 5.10.0-19-amd64

**Comment by rfuchs** [comment]
Correct, the running kernel version doesn't match the kernel headers that are installed. Either reboot into the new (-19) kernel, or install the linux-headers package matching the running kernel.

**Comment by korithi** [comment]
Ok, will try to do that and will keep you posted even for the ticket update.
Thanks!

**Comment by dilyanpalauzov** [comment]
> Nov 08 15:55:50 webrtc rtpengine-iptables-setup[583]: iptables v1.8.7 (nf_tables): unknown option "--id"

There are two iptables - legacy and nft.  You are using legacy, see https://developers.redhat.com/blog/2020/08/18/iptables-the-two-variants-and-their-relationship-with-nftables .

Substitute `iptables -I INPUT -p udp -j RTPENGINE --id 42` with `iptables-nft -I INPUT -p udp -j RTPENGINE --id 42`.

Actually the README files need to be updated to use iptables-nft, I guess it is always a symlink of `iptables`, whenever `--id 42` does not make problems.
